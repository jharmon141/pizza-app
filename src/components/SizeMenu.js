import React from 'react'
import { Col, Row } from 'react-bootstrap'

class SizeMenu extends React.Component {

    componentDidMount() {
        let pizza = this.props.pizzaSizes.find(pizza => {
            return pizza.name === this.props.selectedSize
        })
        this.props.initialSize(pizza)
    }

    render() {
        return (
            <div>
            <Row>
                <Col xs={4} xsOffset={4}>
                    <h2>Choose a size:</h2>
                    <select 
                        onChange={this.props.handleSelectSize} 
                        value={this.props.selectedSize}>

                        {this.props.pizzaSizes.map(size => {
                            return <option 
                                key={size.name} 
                                value={size.name}>
                                {size.name.toUpperCase()}
                                </option>
                        })}
                    </select>
                    <br />
                    <span 
                        className="button" 
                        onClick={this.props.handleSubmitSize}>
                       Continue 
                    </span>
                </Col>
            </Row>
        </div>
        )
    }
}

export default SizeMenu
