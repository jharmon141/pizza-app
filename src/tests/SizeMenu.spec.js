import React from 'react'
import SizeMenu from '../components/SizeMenu'
import renderer from 'react-test-renderer'

describe('SizeMenu Component', () => {
   it('renders without crashing', () => {
      let pizzaSizes = [{name: 'small'}, {name: 'medium'}, {name: 'large'}]

      const tree = renderer.create(
         <SizeMenu 
            pizzaSizes={pizzaSizes}
            initialSize={() => {}}
         />
      ).toJSON()
      expect(tree).toMatchSnapshot()
   })
})
