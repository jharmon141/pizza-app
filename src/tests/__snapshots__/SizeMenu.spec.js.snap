// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeMenu Component renders without crashing 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-4 col-xs-offset-4"
    >
      <h2>
        Choose a size:
      </h2>
      <select
        onChange={undefined}
        value={undefined}
      >
        <option
          value="small"
        >
          SMALL
        </option>
        <option
          value="medium"
        >
          MEDIUM
        </option>
        <option
          value="large"
        >
          LARGE
        </option>
      </select>
      <br />
      <span
        className="button"
        onClick={undefined}
      >
        Continue
      </span>
    </div>
  </div>
</div>
`;
